src/routes/postSkipSegments.ts

```diff
@@ -505,20 +505,21 @@ export async function postSkipSegments(req: Request, res: Response): Promise<Res
         return res.status(userWarningCheckResult.errorCode).send(userWarningCheckResult.errorMessage);
     }
 
-    const isVIP = (await isUserVIP(userID)) || (await isUserTempVIP(userID, videoID));
+    const isVIP = (await isUserVIP(userID));
+    const isTempVIP = (await isUserTempVIP(userID, videoID));
     const rawIP = getIP(req);
 
     const newData = await updateDataIfVideoDurationChange(videoID, service, videoDuration, videoDurationParam);
     videoDuration = newData.videoDuration;
     const { lockedCategoryList, apiVideoDetails } = newData;
 
     // Check if all submissions are correct
-    const segmentCheckResult = await checkEachSegmentValid(rawIP, paramUserID, userID, videoID, segments, service, isVIP, lockedCategoryList);
+    const segmentCheckResult = await checkEachSegmentValid(rawIP, paramUserID, userID, videoID, segments, service, isVIP, isTempVIP, lockedCategoryList);
     if (!segmentCheckResult.pass) {
         return res.status(segmentCheckResult.errorCode).send(segmentCheckResult.errorMessage);
     }
 
-    if (!isVIP) {
+    if (!isVIP || !isTempVIP) {
```

Comment:
This seems like an error: both VIP and tempVIP status would be required at the same time to bypass automod checks
```suggestion
    if (!(isVIP || isTempVIP)) {
```